File: src/test/java/org/micoli/php/JsonToPhpArrayConverterTest.java
package org.micoli.php;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Collection;
import java.util.stream.Collectors;
import java.util.stream.Stream;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;
import org.micoli.php.symfony.cliDumpParser.JsonToPhpArrayConverter;

@RunWith(Parameterized.class)
public class JsonToPhpArrayConverterTest {

    private static final String directory = "src/test/resources/jsonToPhpArrayConverterTestCases/";
    private final String filename;

    public JsonToPhpArrayConverterTest(String filename) {
        this.filename = filename;
    }

    @Parameterized.Parameters(name = "testItParseDump({0})")
    public static Collection<Object[]> data() throws IOException {
        try (Stream<Path> stream = Files.list(Paths.get(directory))) {
            return stream.filter(path -> path.toString().endsWith(".json"))
                    .map(path -> new Object[] {path.getFileName().toString().replace(".json", "")})
                    .collect(Collectors.toList());
        }
    }

    @Test
    public void testItConvertJsonToPhp() throws IOException {
        assert (readFile(".php").equals(JsonToPhpArrayConverter.convertJsonToPhp(readFile(".json"))));
    }

    private String readFile(String path) throws IOException {
        return String.join("\n", Files.readAllLines(Paths.get(directory + filename + path), StandardCharsets.UTF_8));
    }
}

File: src/test/kotlin/org/micoli/php/JsonToPhpArrayConverterTest.kt
package org.micoli.php

import java.io.IOException
import java.nio.charset.StandardCharsets
import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.Paths
import java.util.stream.Collectors
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized
import org.micoli.php.symfony.cliDumpParser.JsonToPhpArrayConverter

@RunWith(Parameterized::class)
class JsonToPhpArrayConverterTest(private val filename: String?) {
    @Test
    @Throws(IOException::class)
    fun testItConvertJsonToPhp() {
        assert(readFile(".php") == JsonToPhpArrayConverter.convertJsonToPhp(readFile(".json")))
    }

    @Throws(IOException::class) private fun readFile(path: String?): String = Files.readAllLines(Paths.get(DIRECTORY + filename + path), StandardCharsets.UTF_8).joinToString("\n")

    companion object {
        private const val DIRECTORY = "src/test/resources/jsonToPhpArrayConverterTestCases/"

        @JvmStatic
        @Parameterized.Parameters(name = "Filter with \"{0}\" and isRegex is {1}")
        @Throws(IOException::class)
        fun parameters(): Collection<Array<String>> {
            Files.list(Paths.get(DIRECTORY)).use { stream ->
                return stream
                  .filter { path: Path? -> path != null }
                  .filter { path: Path -> path.toString().endsWith(".json") }
                  .map { path: Path -> arrayOf(path.fileName.toString().replace(".json", "")) }
                  .collect(Collectors.toList())
            }
        }
    }
}
